### PR TITLE
Assign additional role citizen for documents to overcome Link defendant failure in CCD

### PR DESCRIPTION
**Pull request by dharmendrak**
### Change description ###
Assign additional role citizen for documents to overcome Link defendant failure in CCD

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java [hunk 1]
```diff
@@ -84,7 +84,7 @@ public ClaimDocument uploadDocument(String authorisation, PDF pdf) {
                 authorisation,
                 authTokenGenerator.generate(),
                 userDetails.getId(),
-                asList(userRoles),
+                asList(userRoles.split(",")),
```
Comment (ckalista) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java]: We don't have a test for list of roles?

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java]: I think we could have chosen  `List<String> userRoles ` and bind coma separated properties. Please see example here:
https://github.com/hmcts/cmc-claim-store/blob/master/ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/services/ClaimMigrator.java#L35 .

**Comment by ckalista** [comment]
This is OK as a temporary fix to get us over the line however there is something smelly in the fact we are specifying 2 roles for documents - will chase this.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java (hunk 1 above)
Comment (srineelesh) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java]: userRoles.split(",") -- the split() is not required .. please remove this

File: charts/cmc-claim-store/values.yaml [hunk 2]
```diff
@@ -43,4 +43,4 @@ java:
     RPA_NOTIFICATIONS_RESPONSERECIPIENT: civilmoneyclaims+rpa-defence-response@gmail.com
     RPA_NOTIFICATIONS_COUNTYCOURTJUDGEMENTRECIPIENT: civilmoneyclaims+rpa-county-court-judgement@gmail.com
     RPA_NOTIFICATIONS_PAIDINFULLRECIPIENT: civilmoneyclaims+rpa-paid-in-full@gmail.com
-    DOCUMENT_MANAGEMENT_USERROLES: caseworker-cmc
+    DOCUMENT_MANAGEMENT_USERROLES: caseworker-cmc,citizen
```
Comment (srineelesh) [inline charts/cmc-claim-store/values.yaml]: you might want to update the environment.properties as well

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java [hunk 3]
```diff
@@ -124,6 +125,7 @@ public ClaimDocument logUploadDocumentFailure(
     public byte[] downloadDocument(String authorisation, URI documentSelf, String baseFileName) {
         try {
             UserDetails userDetails = userService.getUserDetails(authorisation);
+            String userRoles = String.join(",", this.userRoles);
```
Comment (srineelesh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java]: this is unnecessary. The previous code was working fine

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java [hunk 4]
```diff
@@ -61,7 +62,7 @@ public DocumentManagementService(
         AuthTokenGenerator authTokenGenerator,
         UserService userService,
         AppInsights appInsights,
-        @Value("${document_management.userRoles}") String userRoles
+        @Value("${document_management.userRoles}") List<String> userRoles
```
Comment (srineelesh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java]: this is unnecessary. The previous code was working fine

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java [hunk 5]
```diff
@@ -84,7 +85,7 @@ public ClaimDocument uploadDocument(String authorisation, PDF pdf) {
                 authorisation,
                 authTokenGenerator.generate(),
                 userDetails.getId(),
-                asList(userRoles),
+                userRoles,
```
Comment (srineelesh) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java]: not sure why this is changed.

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementServiceTest.java [hunk 6]
```diff
@@ -108,16 +118,17 @@ public void shouldDownloadDocumentFromDocumentManagement() {
         URI docUri = URI.create("http://localhost:8085/documents/85d97996-22a5-40d7-882e-3a382c8ae1b4");
 
         when(documentMetadataDownloadClient
-            .getDocumentMetadata(anyString(), anyString(), anyString(), anyString(), anyString())
+            .getDocumentMetadata(anyString(), anyString(), eq(USER_ROLES_JOINED), anyString(), anyString())
```
Comment (srineelesh) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementServiceTest.java]: this is unnecessary. The previous code was working fine

File: charts/cmc-claim-store/values.yaml (hunk 2 above)
Comment (dharmendrak) [reply charts/cmc-claim-store/values.yaml]: thanks for reminding, i'had to take them out

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementServiceTest.java [hunk 7]
```diff
@@ -40,6 +41,9 @@
 @RunWith(SpringJUnit4ClassRunner.class)
 public class DocumentManagementServiceTest {
 
+    public static final ImmutableList<String> USER_ROLES = ImmutableList.of("caseworker-cmc", "citizen");
```
Comment (srineelesh) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementServiceTest.java]: these are redundant

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java (hunk 4 above)
Comment (srineelesh) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java]: we are converting to list here and then again converting to string. Please remove this redundant code.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java (hunk 3 above)
Comment (srineelesh) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java]: we are converting to list above and then again converting to string. Please remove this redundant code.

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementServiceTest.java (hunk 7 above)
Comment (dharmendrak) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementServiceTest.java]: just to verify user roles being used

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementServiceTest.java (hunk 6 above)
Comment (dharmendrak) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementServiceTest.java]: just to verify user roles being used

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java (hunk 1 above)
Comment (ckalista) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java]: We scratched our heads with @srineelesh to understand why removing the `split` works and it indeed does work! The update method interface forces us to supply a list so we are just going to comply to this and provide a list of seperate roles items - so keeping the split here is fine. @sabahirfan 's suggestion is a minor improvement we may want to implement when we roll out a better fox for this. I'll raise a story.

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java]: 👍

Comment (srineelesh) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementService.java]: @ckalista : either ways in the DM client that list is being converted to a comma separate string .. please have a look at https://github.com/hmcts/document-management-client/blob/f6709113c9c056f97ace73d3504951b314a73130/src/main/java/uk/gov/hmcts/reform/document/DocumentUploadClientApi.java#L108 . Please look into the history of this code to understand the issue more.

**Pull request merged**